### ISSUE TITLE
Add Apply All button on particles collection.items

### ISSUE THREAD

**Issue by nicutor** [gantry/gantry5#1287]
Hi guys,

Some of my clients are pissed off because of too many clicks on Apply button from collection items if they want to modify only one nested item. I also realize that they have right.

Here are the steps if let's say we have 3 nested collection items:

1. Clicks on all nested collection items (3 clicks)
2. Modify item
3. Click on all Apply buttons from each item (another 3 clicks)

I think that big improvement will be if there will be an Apply All/Apply and Save button on each collection item, near to Apply button, that will save everything and close the particle window directly from that collection item without to be forced to click on Apply button from all Items.

1. Clicks on all nested collection items (3 clicks)
2. Modify item
3. Click on all Apply All button on modified item (just 1 click)

I guess this is easily to implement because there is already an "Apply and Save" button near to Apply button on Layout > Sections/Particles > Settings

Thank you for consideration on this issue.

**Comment by w00fz** [comment]
What do you mean by nested collection items? Collection list within a collection item?

**Comment by w00fz** [comment]
Maybe I didn't get what you mean but couldn't you just edit all particles at once, make the changes, and apply (2 clicks total)?

![gantry 5 dev - administration 2016-05-17 16-20-34](https://cloud.githubusercontent.com/assets/11734/15342585/63d99ace-1c4b-11e6-900a-5e05af838d95.png)

**Comment by nicutor** [comment]
@w00fz, yes, by nested collection items I mean collection list within a collection item.

Editing all at once is the same thing. If there are other collection list on collection list, and the last one have other collection list on it (and so on) the apply process is a pain (I don't need to count that the missing of TABs to separate the settings its making us to scroll down a lot to press on Apply button).

Entering on those its a must and can't be simplified, but, once the modification is done on the last or intermediary level, an Apply and Save/Apply All button will save everything from 1 shut.

For example, I have a particle with four levels of collection items and after 4 clicks to enter on those, I need to scroll down a lot and click on Apply button by 4 times to save it. Probably other devs didn't created yet a particle with multiple nested collection list, but as some point somebody else will do it and will face with this issue too, like I and my clients do.

As usual, I am adding issues like this in order to improve the end user experience with gantry. I personally didn't found this to be too hard, because I've created the issue and modify it 2 times and that was all. But, after a while when I've asked my clients, this was the first thing they said to me.

And of course I will add multiple issues like this :)

Thank you.

**Comment by w00fz** [comment]
That it a lot of nested collections :D
I see your point though and may be something we look into adding on a future release. 
It's just not as easy as it looks to apply to all, every modal is independent and most times you apply, an Ajax validation begins and could potentially halt the execution in case of error. Automating this will mean having to deal with all of the above without user interaction and things could go bad real quick.
The input from the user is best to deal with problems.

// Djamil Legato // From iPhone //

**Comment by nicutor** [comment]
Ok @w00fz, thank you for this. Don't forget to add a Future Release tag to this issue :)
Have a great day there!
